pkg/provider/provider.go

```diff
@@ -858,3 +878,29 @@ func fetchCertChains(data []byte) ([]byte, error) {
 	}
 	return pemData, nil
 }
+
+// splitCertAndKey takes the given data and splits it into cert and key
+// this function doesn't check if the returned cert and key is not empty as this
+// can't be enforced. It is possible the secret in the key vault only contains the
+// cert or key.
+func splitCertAndKey(certAndKey string) (string, string) {
```

Comment:
Could we add named returned params for better code readability?

Reply:
done!